src/init/register_configure_agent.sh

```diff
@@ -162,14 +206,76 @@ unset_vars() {
     done
 }
 
+# Function to convert strings to lower version
 tolower () {
    echo $1 | tr '[:upper:]' '[:lower:]'
 }
 
-main () {
 
+# Add auto-enrollment configuration block
+add_auto_enrollment () {
+    start_config="$(grep -n "<auto_enrollment>" ${INSTALLDIR}/etc/ossec.conf | cut -d':' -f 1)"
+    end_config="$(grep -n "</auto_enrollment>" ${INSTALLDIR}/etc/ossec.conf | cut -d':' -f 1)"
+    if [ -n "${start_config}" ] && [ -n "${end_config}" ]; then
+        start_config=$(( start_config + 1 ))
+        end_config=$(( end_config - 1 ))
+        sed -n "${start_config},${end_config}p" ${INSTALLDIR}/etc/ossec.conf >> "${TMP_ENROLLMENT}"
+    else
+        # Write the client configuration block
+        echo "<ossec_config>" >> "${TMP_ENROLLMENT}"
+        echo "  <client>" >> "${TMP_ENROLLMENT}"
+        echo "    <enrollment>" >> "${TMP_ENROLLMENT}"
```

Comment:
**delay_after_enrollment** is a setting meant for stressed environments. I think it will be a good idea to include it in the configurable deployment options as it will bring the users the capability to avoid possible problems due to massive automated deployments.